source/core_mqtt.c

```diff
@@ -832,7 +832,7 @@ static int32_t sendMessageVector( MQTTContext_t * pContext,
         }
 
         /* Check for timeout. */
-        if( pContext->getTime() >= timeoutMs )
+        if( calculateElapsedTime( pContext->getTime(), startTime ) > MQTT_SEND_TIMEOUT_MS )
```

Comment:
What happens if `pContext` overflows and wraps? Couldn't this return a negative elapsed time then? Shouldn't that be handled here?

Reply:
Really my core issue here is forgetting that the subtraction will ALWAYS return a positive value given the value is an unsigned integer....which is embarrassing to admit in a persisted form like a Github PR.

Reply:
Making mistakes is only human :)

Thank you for the review @kstribrnAmzn.